compiler/rustc_resolve/src/macros.rs

```diff
@@ -500,6 +502,18 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             Err(Determinacy::Undetermined) => return Err(Indeterminate),
         };
 
+        if kind == MacroKind::Attr
+            && path.segments.len() == 2
```

Comment:
```suggestion
            && path.segments.len() >= 2
```
Looks like `#[diagnostic::foo::bar]` is not reported in any way, but it probably should be.

Technically this condition can be removed entirely because tool attributes are always >= 2 segments.